megamek/src/megamek/server/ServerHelper.java

```diff
@@ -95,6 +95,26 @@ public static void resolveAeroHeat(Game game, Entity entity, Vector<Report> vPha
             r.subject = entity.getId();
             vPhaseReport.add(r);
         }
+        
+        // Add or subtract heat due to extreme temperatures TO:AR p60
+        if (game.getPlanetaryConditions().getTemperatureDifference(50, -30) != 0) {
```

Comment:
I'd suggest not doing this if the game board is in space.

Reply:
Also, I feel like we can just call getTemperatureDifference once instead of four times. Internally, that function is one giant, inefficient WTF, so I'd rather minimize the number of times it's called.

Reply:
Finally, we need to make sure we're not doing any of this if the temp is within "normal" range. Currently, it looks like it'll issue a report even in "normal" temp range, which we want to avoid so that we don't clutter things up any more than we already do.

Reply:
Blame whoever wrote the original code for the mess please lol, this section was a modified copy paste from GameManager line 18481 (Mech heat in extreme temps)

As to questions in order
1. That shouldn't be a problem, have any idea where the variable is set if the map is in space or not? Would help speed up that fix
2. As with my opening statement, this is the original code slightly modified. I can _try_ to clean it up a little.
3. If it doesn't for mechs it shouldn't for aero. I did test to be sure and it does not generate a report in the Heat Phase.

Reply:
Nevermind on one, it was easier to find than I thought. That one is now fixed.

Reply:
I'm not asking you to clean up the getTemperatureDifference function, I'm asking you to call it once and store the results. Right now, you're calling it between three and four times each time through the logic block that you added, whereas you can get away with just once.

Reply:
Ok, misunderstood what you were asking for then. Will have it done tonight, both aero and original mech code.